msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/SignUpEmailPasswordTest.kt

```diff
@@ -93,4 +110,148 @@ class SignUpEmailPasswordTest : NativeAuthPublicClientApplicationAbstractTest()
             }
         }
     }
+
+    /**
+     * Sign up with email + password. Verify email address using email OTP, resend OTP and then set password.
+     * (use case 1.1.5, Test case 16)
+     */
+    @Test
+    fun testSuccessOTPResend() {
+        config = getConfig(defaultConfigType)
+        application = setupPCA(config, defaultChallengeTypes)
+
+        retryOperation {
+            runBlocking { // Running with runBlocking to avoid default 10 second execution timeout.
+                val user = tempEmailApi.generateRandomEmailAddress()
+                val signUpResult = application.signUp(user)
+                assertResult<SignUpResult.CodeRequired>(signUpResult)
+
+                val resendCodeResult = (signUpResult as SignUpResult.CodeRequired).nextState.resendCode()
+                assertResult<SignUpResendCodeResult.Success>(resendCodeResult)
+
+                val otp = tempEmailApi.retrieveCodeFromInbox(user)
+                val submitCodeResult = (resendCodeResult as SignUpResendCodeResult.Success).nextState.submitCode(otp)
+                assertResult<SignUpResult.PasswordRequired>(submitCodeResult)
+
+                val submitPasswordResult = (submitCodeResult as SignUpResult.PasswordRequired).nextState.submitPassword(getSafePassword().toCharArray())
+                Assert.assertTrue(submitPasswordResult is SignUpResult.Complete)
+            }
+        }
+    }
+
+
+    /**
+     * Sign up with email + password. Resend email OOB.
+     * (use case 1.1.2, Test case 26)
+     */
+    @Test
+    fun testResendEmailOOB() {
+        config = getConfig(defaultConfigType)
+        application = setupPCA(config, defaultChallengeTypes)
+
+        runBlocking { // Running with runBlocking to avoid default 10 second execution timeout.
+            val user = tempEmailApi.generateRandomEmailAddress()
+            val signUpResult = application.signUp(user)
+            assertResult<SignUpResult.CodeRequired>(signUpResult)
+            val codeRequiredState = (signUpResult as SignUpResult.CodeRequired).nextState
+            val resendCodeResult = codeRequiredState.resendCode()
+            assertResult<SignUpResendCodeResult.Success>(resendCodeResult)
+        }
+    }
+
+    /**
+     * Sign up with email + password. User already exists with given email as email-pw account.
+     * (use case 1.1.10, Test case 28)
+     */
+    @Test
+    fun testErrorUserExistAsPassword() {
+        config = getConfig(ConfigType.SIGN_IN_PASSWORD)
+        application = setupPCA(config, defaultChallengeTypes)
+
+        runBlocking { // Running with runBlocking to avoid default 10 second execution timeout.
+            val user = config.email
+            val signUpResult = application.signUp(user)
```

Comment:
Password is missing.

Same in tests 1.1.2, 1.1.11 and 1.1.12

Reply:
Nice catch. Done.